test/logger.utils.spec.ts

```diff
@@ -0,0 +1,32 @@
+import { checkPattern } from '../src';
+
+describe('logger utils', () => {
+  describe('checkPattern', () => {
+    const str = 'quick fox jumps over the lazy dog';
+    it('should not match undefined or null', () => {
+      expect(checkPattern(str, undefined)).toBe(false);
+      expect(checkPattern(str, null)).toBe(false);
+    });
+
+    it('should match string pattern', () => {
+      expect(checkPattern(str, 'quick')).toBe(false);
+      expect(checkPattern(str, str)).toBe(true);
+    });
+
+    it('should match RegExp pattern', () => {
+      expect(checkPattern(str, /fox/)).toBe(true);
+      expect(checkPattern(str, /^fox/)).toBe(false);
+      expect(checkPattern(str, /dog$/)).toBe(true);
+    });
+
+    it('should match string or RegExp patterns', () => {
```

Comment:
data driven style can be used to simplify the test
https://jestjs.io/docs/api#testeachtablename-fn-timeout